frontend/src/components/map/LeafletMapContainer.tsx

```diff
@@ -16,49 +17,45 @@ export interface MapProps {
   const locations = useMultipleLocationsByIdOrNull(locationIds);
   const usersPosition = useCurrentLocation();
-  const [zoom, setZoom] = useState<number>();
+  const [view, setView] = useState<View>({ bounds: NRW_BOUNDS });
 
-  const lat = usersPosition.state === "located" ? usersPosition.location.coords.latitude : 51.183334;
-  const lng = usersPosition.state === "located" ? usersPosition.location.coords.longitude : 7.200000;
+  const bounds: LatLngTuple[] = locations.map((location) => [location.details.coordinates.latitude, location.details.coordinates.longitude]);
+  const markers = locations.map((location, index) => <MapMarker key={"marker" + index} origin={origin} destination={destination} location={location} onItemClicked={onItemClicked} />);
 
-  const getLocationsCoords = (): LatLngTuple[] => locations
-    .map((location) => [location.details.coordinates.latitude, location.details.coordinates.longitude]);
+  useEffect(() => {
+    if (usersPosition.state === "located") {
+      const lat = usersPosition.location.coords.latitude;
+      const lng = usersPosition.location.coords.longitude;
 
-  /* const getBounds = (): LatLngTuple[] => {
-    const bounds = getLocationsCoords();
-    if (bounds.length <= 0) {
-      return [[currentLocation.details.latitude ?? 51.183334, currentLocation.details.longitude ?? 7.200000]];
+      setView({ center: [lat, lng], zoom: 15 });
     }
-    return bounds;
-  }; */
-
-  const renderMarker = (): ReactElement[] => locations.map((location, index) => <MapMarker key={"marker" + index} origin={origin} destination={destination} location={location} onItemClicked={onItemClicked} />);
-
-  const getPolygonPositions = getLocationsCoords;
-
-  useEffect(() => {
-    setZoom(15);
   }, []);
 
-  return zoom ? (
-    <MapContainer id="map"
-      center={[lat, lng]}
-      zoom={zoom}>
-      <TileLayer
-        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-      />
-      {/* <MapController bounds={getBounds()} /> */}
-      {renderMarker()}
-      {<CurrentLocationMapMarker />}
-      {showLines
-        ? <Polygon color={"rgb(77, 77, 77)"} opacity={1} dashArray={"20,15"} weight={2} positions={getPolygonPositions()} />
-        : <></>}
-    </MapContainer>
-  ) : <div />;
+  useEffect(() => {
+    if (bounds.length > 0) {
+      setView({ bounds });
+    }
+  }, [locations]);
+
+  return <ReactiveMapContainer
+    style={{ height: "100%", width: "100%" }}
+    view={view}
+  >
+    <TileLayer
+      attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+      url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+    />
+    {markers}
+    <CurrentLocationMapMarker />
+    {showLines
+      ? <Polygon color={"rgb(77, 77, 77)"} opacity={1} dashArray={"20,15"} weight={2} positions={bounds} />
+      : <></>}
```

Comment:
```suggestion
    {showLines && <Polygon color={"rgb(77, 77, 77)"} opacity={1} dashArray={"20,15"} weight={2} positions={bounds} />}
```

Deja Vu?

![whoa-deja-vu-matrix-glitch](https://github.com/Oeffis/Oeffis/assets/4148404/6b8f7eaf-ea2c-4858-9c46-8aa9cb7adf50)